storage/cloudspanner/tree_storage.go

```diff
@@ -208,12 +209,8 @@ func (t *treeTX) currentSTH(ctx context.Context) (*spannerpb.TreeHead, error) {
 }
 
 func (t *treeTX) writeRev(ctx context.Context) (int64, error) {
-	if err := t.getLatestRoot(ctx); err == storage.ErrTreeNeedsInit {
-		return 0, nil
-	} else if err != nil {
-		return -1, fmt.Errorf("writeRev(): %v", err)
-	}
-	return t._writeRev, nil
+	err := t.getLatestRoot(ctx)
```

Comment:
This looks as though it may break tree creation due to removing the special case when a tree exists, but has no root?

Reply:
This is where the new integration tests help us :-)
The `ErrTreeNeedsInit` error was being special cased a few *too* many times. It's special cased both here and in `getLatestRoot`. The result is that the second special case overrides the first, causing the first revision to be 0 rather than -1.

Reply:
I take this back -- another nice reason to have integration tests :-)